cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerKerberosService.java

```diff
@@ -62,7 +64,16 @@ public void setupKerberos(ApiClient client, HttpClientConfig clientConfig, Stack
             ApiCommand importAdminCredentials =
                     clouderaManagerResourceApi.importAdminCredentials(kerberosConfig.getPassword(), kerberosConfig.getPrincipal());
             clouderaManagerPollingServiceProvider.kerberosConfigurePollingService(stack, client, importAdminCredentials.getId());
+        }
+    }
+
+    public void configureKerberosViaApi(ApiClient client, HttpClientConfig clientConfig, Stack stack, ClouderaManagerRepo clouderaManagerRepoDetails)
+            throws ApiException, CloudbreakException {
+        Cluster cluster = stack.getCluster();
+        if (!CMRepositoryVersionUtil.isEnableKerberosSupportedViaClusterDefinition(clouderaManagerRepoDetails)
```

Comment:
I think this condition should be at the method invocation, not within the method. Then it will clearly show when we're doing it with API.